### PR TITLE
docs: new architecture diagram

### LINKED ISSUE #2430: Update SaaS architecture diagram
[This diagram](https://docs.flagsmith.com/version-comparison#saas-architecture) needs some updates:

 - Add redis for real time
 - Remove / clarify dashboard - currently says that it resides in AWS London. Maybe this is true, but we should probably denote that it's not part of our AWS account?

### PR DESCRIPTION
**Pull request by dabeeeenster**
Fixed #2430

### CODE REVIEW

File: docs/docs/system-administration/architecture.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+title: Architecture
+---
+
+## Self Hosted / On Prem
+
+![Image](/img/self-hosted-architecture.svg)
+
+## SaaS
+
+![Image](/img/saas-architecture.svg)
```
Comment (matthewelwell) [inline docs/docs/system-administration/architecture.md:11]: Looks like the updated diagram for the SaaS architecture is not part of this PR yet?

File: docs/docs/system-administration/architecture.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+title: Architecture
```
Comment (matthewelwell) [inline docs/docs/system-administration/architecture.md:2]: I'm not sure that I totally agree that this belongs in the 'System Administration' section... but I don't know that I have a better suggestion without creating a totally new section in the nav.

File: docs/docs/system-administration/architecture.md (hunk 1 above)
Comment (matthewelwell) [reply docs/docs/system-administration/architecture.md:11]: image

File: docs/docs/system-administration/architecture.md (hunk 2 above)
Comment (dabeeeenster) [reply docs/docs/system-administration/architecture.md:2]: Yeah it has never really had anywhere decent to live. I think this is the least worst option

File: docs/docs/system-administration/architecture.md (hunk 1 above)
Comment (dabeeeenster) [reply docs/docs/system-administration/architecture.md:11]: Oops fixed

**Review by matthewelwell — APPROVED** [review]
